### PR TITLE
feat(tab): replace React Hook functions with SDK package

### PR DESCRIPTION
**Pull request by tecton**
Replace `useData`, `useGraph`, `useTeamsFx` with the ones in @microsoft/teamsfx-react.

Changed samples:
- hello-world-tab-with-backend
- hello-world-tab
- hello-world-tab-without-sso

Related: https://github.com/OfficeDev/TeamsFx/pull/4315

### CODE REVIEW

File: hello-world-tab/tabs/package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
         "@microsoft/microsoft-graph-client": "^3.0.1",
         "@microsoft/teams-js": "^1.9.0",
         "@microsoft/teamsfx": "^0.6.0",
+        "@microsoft/teamsfx-react": "0.0.2-alpha.a4c0338b1.0",
```
Comment (tecton) [inline hello-world-tab/tabs/package.json:13]: Will change to stable version

**Review by KennethBWSong — APPROVED** [review]
approve for non sso tab sample

File: hello-world-tab-with-backend/tabs/src/components/sample/AzureFunctions.tsx [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 import { TeamsFx } from "@microsoft/teamsfx";
+import { TeamsFxContext } from "../Context";
 
-var functionName = process.env.REACT_APP_FUNC_NAME || "myFunc";
+const functionName = process.env.REACT_APP_FUNC_NAME || "myFunc";
+let teamsfx: TeamsFx | undefined;
 
 async function callFunction() {
+  if (!teamsfx) {
+    return;
```
Comment (SLdragon) [inline hello-world-tab-with-backend/tabs/src/components/sample/AzureFunctions.tsx:13]: Do we need to show warning or alert to tell user teamsfx is not initialized?

Comment (tecton) [reply hello-world-tab-with-backend/tabs/src/components/sample/AzureFunctions.tsx:13]: Thanks, I'll add code that throws error here.

**Review by SLdragon — APPROVED** [review]

**Pull request merged**